From 4162e13b06a6d8b0c640be7a4398190d6eeeae5d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ma=C3=ADra=20Canal?= <maira.canal@usp.br>
Date: Fri, 12 Aug 2022 14:31:57 -0300
Subject: [PATCH] add trigger to digital channels and add more commands to tektronix

---
 ivi/tektronix/tektronixBaseScope.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ivi/tektronix/tektronixBaseScope.py b/ivi/tektronix/tektronixBaseScope.py
index b1e2167..b19b2b0 100644
--- a/ivi/tektronix/tektronixBaseScope.py
+++ b/ivi/tektronix/tektronixBaseScope.py
@@ -146,6 +146,10 @@ MeasurementFunctionMappingDigital = {
         'period': 'period',
         'width_negative': 'nwidth',
         'width_positive': 'pwidth',
+        'edgecount_negative': 'nedgecount',
+        'edgecount_positive': 'pedgecount',
+        'pulsecount_negative': 'npulsecount',
+        'pulsecount_positive': 'ppulsecount',
         'duty_cycle_positive': 'dutycycle'}
 ScreenshotImageFormatMapping = {
         'tif': 'tiff',
@@ -434,6 +438,7 @@ class tektronixBaseScope(scpi.common.IdnCommand, scpi.common.Reset, scpi.common.
             for i in range(self._digital_channel_count):
                 self._channel_name.append("d%d" % i)
                 self._channel_label.append("")
+                self._channel_trigger_level.append(0.0)
                 self._digital_channel_name.append("d%d" % i)
 
             for i in range(self._analog_channel_count, self._channel_count):
-- 
2.37.1

